src/test/aws/construct/static-asset-deployment.test.ts

```diff
@@ -22,91 +19,202 @@ const testStackProps = {
   stackName: 'test',
   stage: 'test',
   stageContextPath: 'src/test/aws/common/cdkEnv',
-}
-
-class TestCommonStack extends CommonStack {
-  declare props: TestStackProps
+} as any as AppProps
 
-  constructor(parent: cdk.App, name: string, props: cdk.StackProps) {
-    super(parent, name, props)
+const defaultProps = {
+  staticAssetBucket: {
+    bucketName: 'site',
+    autoDeleteObjects: true,
+    removalPolicy: 'destroy',
+    serverAccessLogsPrefix: 'logs/',
+    websiteIndexDocument: 'index.html',
+    websiteErrorDocument: 'index.html',
+    existingBucket: false,
+  },
+  staticAssetDeployment: {
+    prune: true,
+    retainOnDelete: false,
+  },
+  staticAssetsForExport: [{ key: 'myCSV', value: 'test.csv' }],
+} as any as StaticAssetDeploymentProps
 
-    this.construct = new TestStaticAssetDeployment(this, testStackProps.name, this.props)
-  }
+type StaticAssetDeploymentExpected = {
+  bucketCount: number
+  prune: boolean
+  sourceObjectKeys: Array<string>
+  destinationKeyPrefix: string | Matcher
+  distributionId: string | Matcher
+  destinationBucket?: string
+}
 
-  protected determineConstructProps(props: cdk.StackProps) {
-    return {
-      ...super.determineConstructProps(props),
-      staticAssetBucket: this.node.tryGetContext('siteBucket'),
-      staticAssetDeployment: {
-        prune: true,
-        retainOnDelete: false,
-      },
-      staticAssetSources: [s3deploy.Source.asset('src/test/aws/common/resources/')],
-      staticAssetsForExport: [{ key: 'myCSV', value: 'test.csv' }],
-    }
-  }
+const defaultExpected: StaticAssetDeploymentExpected = {
+  bucketCount: 1,
+  prune: true,
+  sourceObjectKeys: ['resources.zip'],
+  destinationKeyPrefix: Match.absent(),
+  distributionId: Match.absent(),
 }
 
-class TestStaticAssetDeployment extends StaticAssetDeployment {
-  declare props: TestStackProps
+describe('StaticAssetDeployment', () => {
+  beforeAll(s3AssetSpySetup)
 
-  constructor(parent: Construct, id: string, props: TestStackProps) {
-    super(parent, id, props)
-    this.props = props
-    this.id = 'test-static-asset-deployment'
-    this.initResources()
-  }
-}
+  afterAll(s3AssetSpyRestore)
 
-const app = new cdk.App({ context: testStackProps })
-const stack = new TestCommonStack(app, 'test-static-asset-deployment-stack', testStackProps)
-const template = Template.fromStack(stack)
+  describe.each<TableTestTuple<StaticAssetDeploymentProps, StaticAssetDeploymentExpected>>([
+    ['create bucket when it not existing bucket', defaultProps, defaultExpected],
```

Comment:
@kamaz Can you please align tests to the way it's defined everywhere else? Please raise a backlog with any new proposal with examples

Reply:
https://gradientedge.atlassian.net/browse/MCA-611